File: CommunicationModel/src/BrokerModels/CollectorAccessEvent.cs
using System;

namespace CommunicationModel.BrokerModels
{
	public class CollectorAccessEvent : ServiceEvent
	{
		public string method { get; set; }
		public string requestPath { get; set; }
		public string query { get; set; }

		public string sourceAddr { get; set; }
		public int? sourcePort { get; set; }

		public DateTime requestReceivedTime { get; set; }
		public DateTime responseSendTime { get; set; }

		public int statusCode { get; set; }
		public string responseType { get; set; }
		public long? responseLength { get; set; }

		public CollectorAccessEvent(
			string serviceId,
			string method,
			string requestPath,
			string query,
			string sourceAddr,
			int? sourcePort,
			DateTime requestReceivedTime,
			int statusCode,
			string responseType,
			long? responseLength,
			string additionalDesc = "")

			: base(serviceId, ServiceType.DataCollector, additionalDesc)
		{
			this.method = method;
			this.requestPath = requestPath;
			this.query = query;
			this.sourceAddr = sourceAddr;
			this.sourcePort = sourcePort;
			this.requestReceivedTime = requestReceivedTime;
			this.responseSendTime = DateTime.Now;
			this.statusCode = statusCode;
			this.responseType = responseType;
			this.responseLength = responseLength;
		}
	}
}

File: DashboardClient/TBoardRabbitAdapter/src/main/java/broker/messages/incoming/CollectorAccessEvent.java
package broker.messages.incoming;

import java.util.Date;

public class CollectorAccessEvent extends ServiceEvent {

	public String method;
	public String requestPath;
	public String query;

	public String sourceAddr;
	public int sourcePort;

	public Date requestReceivedTime;
	public Date responseSendTime;

	public int statusCode;
	public String responseType;
	public long responseLength;

	public CollectorAccessEvent(String sourceId,
			ServiceType type,
			String customMessage,
			String method,
			String requestPath,
			String query,
			String sourceAddr,
			int sourcePort,
			Date requestReceivedTime,
			Date responseSendTime,
			int statusCode,
			String responseType,
			long responseLength) {

		super(sourceId, type, customMessage);

		this.method = method;
		this.requestPath = requestPath;
		this.query = query;
		this.sourceAddr = sourceAddr;
		this.sourcePort = sourcePort;
		this.requestReceivedTime = requestReceivedTime;
		this.responseSendTime = responseSendTime;
		this.statusCode = statusCode;
		this.responseType = responseType;
		this.responseLength = responseLength;
	}

}
